gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+        for arg in process.cmdline():
+            if ":" in arg:
+                if IP_PORT_RE.match(arg):
+                    continue
+
+                return append_python_module_to_proc_wd(process, arg)
+
+        logger.warning(f"GunicornApplicationSeparator: matched against process {process} but couldn't find WSGI module")
+        return None
+
+
+class UwsgiApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "uwsgi"
+
+    def is_supported(self, process: Process) -> bool:
+        return process.cmdline()[0] == "uwsgi"
+
+    @staticmethod
+    def _find_wsgi_from_config_file(process: Process) -> Optional[str]:
+        config_file = None
+        for arg in process.cmdline():
+            if arg.endswith(".ini"):
+                config_file = arg
+
+        if config_file is None:
+            return None
+
+        if not os.path.isabs(config_file):
+            config_file = os.path.join(process.cwd(), config_file)
+
+        config = configparser.ConfigParser()
+        config.read(resolve_host_path(process, config_file))
+        try:
+            return config.get("uwsgi", "module")
+        except (configparser.NoSectionError, configparser.NoOptionError):
+            pass
+
+        return None
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        wsgi = get_positional_arg(process.cmdline(), "-w")
+        if wsgi is not None:
+            return wsgi
+
+        wsgi = self._find_wsgi_from_config_file(process)
+        if wsgi is not None:
+            return wsgi
+
+        logger.warning("Couldn't find uwsgi wsgi module, both from cmdline and from config file")
+        return None
+
+
+class CeleryApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "celery"
+
+    def is_supported(self, process: Process) -> bool:
+        if process.cmdline()[0] == "celery":
+            return True
+
+        return len(process.cmdline()) >= 3 and ["-m", "celery"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        app_name = get_positional_arg(process.cmdline(), "-A") or get_positional_arg(process.cmdline(), "--app")
+        if app_name is None:
+            logger.warning("Couldn't find positional argument -A or --app for application indication")
+            return None
+
+        return app_name
+
+
+class PySparkApplicationSeparator(ApplicationSeparator):
+    @property
+    def app_prefix(self) -> str:
+        return "pyspark"
+
+    def is_supported(self, process: Process) -> bool:
+        # We're looking for pythonXX -m pyspark.daemon
+        return process.cmdline() >= 3 and ["-m", "pyspark.daemon"] == process.cmdline()[1:]
+
+    def get_application_name(self, process: Process) -> Optional[str]:
+        return "generic pyspark.daemon"
```

Comment:
Pyspark applications don't have their own name? If they don't I would say the application name should be just "pyspark".  The extra `"generic pyspark.daemon"` is confusing.

Reply:
They do have their own name but unfortunately, extracting it is not simple. I will change to "pyspark"

Reply:
@guybortnikov , can you comment as a TODO:
1. That's it's hard and that's why we don't do it 
2. Any reference to a resource / ... on how to do it? Some pointers for the next guy working on it.